don't include it if we're not using it

also a workaround for loongarch, highway.h fails to include

--- a/numpy/_core/src/umath/loops_trigonometric.dispatch.cpp
+++ b/numpy/_core/src/umath/loops_trigonometric.dispatch.cpp
@@ -3,8 +3,10 @@
 #include "loops_utils.h"
 
 #include "simd/simd.h"
+#if NPY_SIMD_FMA3
 #include <hwy/highway.h>
 namespace hn = hwy::HWY_NAMESPACE;
+#endif
 
 /*
  * Vectorized approximate sine/cosine algorithms: The following code is a
--- a/numpy/_core/src/npysort/highway_qsort.hpp
+++ b/numpy/_core/src/npysort/highway_qsort.hpp
@@ -2,14 +2,20 @@
 #define NUMPY_SRC_COMMON_NPYSORT_HWY_SIMD_QSORT_HPP
 
 #define VQSORT_ONLY_STATIC 1
+#ifdef __loongarch__
+#define NPY_DISABLE_HIGHWAY_SORT
+#else
 #include "hwy/highway.h"
 #include "hwy/contrib/sort/vqsort-inl.h"
+#endif
 
 #include "common.hpp"
 
+#ifndef NPY_DISABLE_HIGHWAY_SORT
 #if !VQSORT_COMPILER_COMPATIBLE
 #define NPY_DISABLE_HIGHWAY_SORT
 #endif
+#endif
 
 #ifndef NPY_DISABLE_HIGHWAY_SORT
 namespace np { namespace highway { namespace qsort_simd {
